### PR TITLE
Make feature gate threadsafe

### LINKED ISSUE #51548: pull-kubernetes-unit TestCRD is flaky
/kind bug

sample from clicking around:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/51448/pull-kubernetes-unit/50367/
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/51539/pull-kubernetes-unit/50354/
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/51479/pull-kubernetes-unit/50345/

### PR DESCRIPTION
**Pull request by liggitt**
Fixes #51548 caused by #51436

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go [hunk 1]
```diff
@@ -197,7 +212,9 @@ func (f *featureGate) Enabled(key Feature) bool {
 
 // AddFlag adds a flag for setting global feature gates to the specified FlagSet.
 func (f *featureGate) AddFlag(fs *pflag.FlagSet) {
+	f.lock.Lock()
 	f.closed = true
+	f.lock.Unlock()
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go]: I strongly prefer the `defer` to unlock.  Some reason I can't?  I'm betting something else is trying to grab it underneath.  I really hate the golang sync library.

**Comment by deads2k** [comment]
question, lgtm otherwise

File: staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go]: both KnownFeatures and the flag set up make use of methods that RLock

**Comment by liggitt** [comment]
@k8s-oncall, tests are [green](https://k8s-gubernator.appspot.com/pr/51707), not sure why the successful kops run wasn't reported

**Pull request merged**